### PR TITLE
Rename package and update dependencies

### PR DESCRIPTION
**Pull request by laulopezreal**
This PR aim is to address https://github.com/biocypher/open-targets/issues/19.

# Renaming
- Folder `otar_biocypher` renamed to `open_targets`
- Module `otar_biocypher_adapter.py` renamed to `open_targets.py`
- Module in folder `scripts` renamed to `open_targets_biocypher_run.py`

# Include the project directory in pyproject.toml
By adding line 6 in `pyproject.toml`, the scripts across the repository can import the content of the `open_targets` folder.

# Dependency Updates:
Updated dependencies in the `poetry.lock` file to ensure compatibility and security.
- `pyspark` to version `3.5.3`
- `pystow` to version `0.5.6`
- `requests`  to version `2.32.3` 
-`pydantic` to version `2.9.2` 
- `biocypher` to version `0.6.1`
- `bioregistry` to version `0.11.22`

# Other minor changes
- Update imports and documentation
- Fix minor code formatting warnings

### CODE REVIEW

**Comment by kpto** [comment]
@slobentanzer According to #19 , this also implies major version bump so I think we should make the decision carefully. How would the breaking change affect our community?

**Comment by laulopezreal** [comment]
> @slobentanzer According to #19 , this also implies major version bump so I think we should make the decision carefully. How would the breaking change affect our community?

Maybe you can create a development/release-protected branch to accumulate some more changes and do a major release at once?

**Comment by kpto** [comment]
@laulopezreal That works, I have discussed with @slobentanzer that we will create a 0.4.0 version milestone that will includes:
1. This PR
2. Refactoring mentioned in https://github.com/biocypher/open-targets/issues/23
3. An alternative query engine to replace pyspark

While the issue #19 mentions a major version bump, 1.0.0 marks the beginning of the public API so we need to carefully determine our requirements first. I will create a branch named 0.4.0 soon then you can change you merge target to it.

File: pyproject.toml
Comment (kpto) [inline pyproject.toml:6]: I think this is a building argument and does not do what you think it does because the lock file has no change related to it. I think the script just works without this and the original workaround. I tested the `main` branch with the workaround deleted and it worked. I don't why it didn't work but I also don't think this is fixing anything as well.

I think we better remove it until we are sure how we want to distribute the adapter.

File: scripts/open_targets_biocypher_run.py
Comment (kpto) [inline scripts/open_targets_biocypher_run.py:6]: As my another comment mentioned, I think we can simply remove the workaround here and the relevant comment block.

File: pyproject.toml
Comment (slobentanzer) [reply pyproject.toml:6]: Agree; we are not building the package to distribute at the moment. While this could be an avenue, we haven't gone so far as to put any adapter on PyPI yet. If we create a package, I am thinking it would be more a central one that contains multiple standard adapters, not each one individually.

File: scripts/open_targets_biocypher_run.py
Comment (slobentanzer) [reply scripts/open_targets_biocypher_run.py:6]: Could be, I haven't tested that recently.

**Comment by kpto** [comment]
> Hi @kpto,
> 
> That is awesome :) it makes a lot of sense to me. Thank you so much, I just changed the target branch. I am happy to chat about how you think I can best contribute to/ help with the other two changes

@laulopezreal Your attitude is much appreciated : )

I am testing duckdb and Polars to see which one is more suitable in our application especially their capability on streaming because Open Targets data is huge and streaming is preferred to remove the memory stress to make our pipeline also useful to users who have a more limited computer.

In the mean time it would be great if you could help on https://github.com/biocypher/open-targets/issues/23. A simple start could be just moving enum classes and helper functions/classes outside the main adapter class to be separated files so the adapter class is cleaner.

File: pyproject.toml
Comment (kpto) [reply pyproject.toml:6]: @laulopezreal According to poetry doc, the default mode is already package-mode so the project is installed in editable mode and I can confirm that with branch `main` I see this in a newly created virtual environment containing this package manifest:
```json
{"dir_info": {"editable": true}, "url": "file:///C:/Users/Paul/Documents/GitHub/open-targets"}
```

Adding this line produces the same package manifest so I don't think your change here matters.

Comment (slobentanzer) [reply pyproject.toml:6]: @kpto, thanks for the explanation. Do we actually need an editable installation?

Comment (kpto) [reply pyproject.toml:6]: @slobentanzer I think it depends how we want to distribute it and we don't need to make our decision now. I just want to point out that the change here is redundant and should be removed.

Comment (slobentanzer) [reply pyproject.toml:6]: While it is redundant, it is often preferred to be explicit. The configuration entry at least leaves no question of the install behaviour. If it really does not cause harm anywhere, I would still keep it, as it more explicitly states what the package install is doing.

I am again coming from the angle of the not-very-experienced programmer type user that we see often in our audience.

Comment (kpto) [reply pyproject.toml:6]: @slobentanzer I meant redundant in terms of @laulopezreal 's intention, it does mean something for project packaging and distribution I think and it is a decision that I don't think we should make here right now. As shown above, the produced package manifest of the local editable installation is the same regardless the value of this argument.

Comment (slobentanzer) [reply pyproject.toml:6]: OK I guess I don't grasp the distinction; I'll leave this to the two of you to sort out. :)

Comment (slobentanzer) [reply pyproject.toml:6]: Just as a note on the Poetry side of things: if we want to install in non-package-mode, we get a warning in the new Poetry version that will become an error soon. Thus, it would be best practice to include this field sooner rather than later, regardless of which value we choose. Here's the message from the `meta-graph` pipeline:

```
Warning: The current project could not be installed: No file/folder found for package biocypher-meta-graph
If you do not want to install the current project use --no-root.
If you want to use Poetry only for dependency management but not for packaging, you can disable package mode by setting package-mode = false in your pyproject.toml file.
In a future version of Poetry this warning will become an error!
```

While this does not occur if we actually do have a package (as in this case), this is not necessary for a BioCypher pipeline. You can have a pipeline without a package. In all these cases, this error will be relevant.

Comment (laulopezreal) [reply pyproject.toml:6]: Hello folks :) Thank you so much for looking into this and giving feedback. I am more inclined towards @slobentanzer view of leaving it there. If I were a user, I would understand better the installation procedure if I see that line.
I see both of your points though. The only reason I included it in the first place, is that in my projects, I like to use a `src` folder:
```
my_repository
├───readme.md
│   
├───barconfig
│    bar.xml
├───src
│   ├───my_module1
│   │    python_script.py
│   └───my_module2 (this is not always the case)
```

In this structure, I need to tell poetry where the source code lives so I can import functions or classes between module1 and module2.

Given the structure of this project, this line is not required because it works with the default configuration in poetry. But is good to keep in mind that it is not always the case and you will find projects where that line would be required.

Hope it makes sense!

Comment (laulopezreal) [reply pyproject.toml:6]: Having said that... this is not a fight that I am going to fight, I understand your view and I believe that it is not incorrect to remove that line. Will change it :) @kpto

**Comment by laulopezreal** [comment]
> > Hi @kpto,
> > That is awesome :) it makes a lot of sense to me. Thank you so much, I just changed the target branch. I am happy to chat about how you think I can best contribute to/ help with the other two changes
> 
> @laulopezreal Your attitude is much appreciated : )
> 
> I am testing duckdb and Polars to see which one is more suitable in our application especially their capability on streaming because Open Targets data is huge and streaming is preferred to remove the memory stress to make our pipeline also useful to users who have a more limited computer.
> 
> In the mean time it would be great if you could help on #23. A simple start could be just moving enum classes and helper functions/classes outside the main adapter class to be separated files so the adapter class is cleaner.

Of course, I actually have started on this in my fork, so I will open the PR here :) I wanted to wait for this PR's resolution and have some feedback from you before opening a new one.

File: pyproject.toml
Comment (kpto) [reply pyproject.toml:6]: @slobentanzer @laulopezreal My main concern was that I was not sure what it is doing and I don't like making changes that I don't understand. The poetry documentation is unclear but as it got attention from you all already so I spent some time on experimenting it and here is my findings:

### Overview
1. This is a packaging argument. While local editable installation tries to simulate the end result, it's not always successful and often inaccurate.
2. There is a standard poetry project structure, this argument is redundant if the our structure follows it. The structure is basically having a directory matching the name property in `pytoml` so this PR has already made it matching. Apart from the common `src` structure which requires this argument, this argument is more for namespace fine tuning.

### `from` and `to` argument
Given this project structure, I intentionally make it complicated to show my findings.
```bash
project_dir
├───mymodule_a
│   └───__init__.py
└───src_b
    └───mymodule_b
        └───__init__.py
```
and this argument value:
```python
packages = [
    {include = "mymodule_a", to = "mynamespace"},
    {include = "mymodule_b", from = "src_b"},
]
```
You end up having this structure after `poetry build` and `pip install`
```bash
site-packages
├───mynamespace
│   └───mymodule_a
│       └───__init__.py
└───mymodule_b
    └───__init__.py
```
so you can import these two modules like this, which makes sense to me
```python
import mynamespace.mymodule_a # expected usage
import mymodule_b # expected usage
import mymodule_a # import error which is correct
```

But when installed locally for editing, poetry tries to mimic this but not very successful, it generates two paths:
```
....../project_dir
....../project_dir/src_b
```
Since poetry cannot alter the directory structure of the project, the ultimate behavors becomes:
```python
import mynamespace.mymodule_a # expected usage, but errored because this directory does not exist
import mymodule_b # expected usage
import src_b.mymodule_b # should error but found when searching ....../project_dir
```

### Conclusion
While the change here is not harmful, I still think it should be removed because it's merely repeating the default behavior and it is hard to explain it's intention. Also according to software engineering principles `KISS` and `YAGNI`, such a default behavior overriding argument should be used only when necessary. When creating a new project using `poetry new`, the generated `pytoml` also does not have this argument. We probably will discuss more in depth about how we want to distribute the project and may need it soon though.

@slobentanzer The error you are seeing is because that you project name `biocypher-meta-graph` does not match the module name `meta_graph`. A side note, poetry is smart enough to map `-` to `_` so you don't need to use underscore on project name.

Comment (slobentanzer) [reply pyproject.toml:6]: Thanks @kpto, that is very useful; I understand this much better now. Agree re the software development principles. If we ever have the need to adjust the default behaviour, we can introduce the parameter at this point. I try to help inexperienced users, but you are correct that this help should not be confusing or beside the point. Maybe we find a way to introduce this in the docs somewhere, so we can at least point the users to a solution or reason.

Comment (laulopezreal) [reply pyproject.toml]: @kpto wow thank you so much for investigating for sharing your findings. I feel that just now I fully understand what poetry does under the hood. Very valuable information as this is something I use recurrently on my projects. 

I've pushed the agreed changes so, hopefully, the PR is okay now. Let me know otherwise :) and thanks again for such an interesting discussion, guys!

File: scripts/open_targets_biocypher_run.py
Comment (laulopezreal) [reply scripts/open_targets_biocypher_run.py]: Resolving this conversation: I've removed the comments because we now know what was going on.

**Review by kpto — APPROVED** [review]
I am not sure the distribution name should be `open_target` rather than `open-target`. It seems dash is more common but underscore is accepted as well, though all symbols are normalised to `-` when making a name comparison as if like `-` is preferred as documented here https://packaging.python.org/en/latest/specifications/name-normalization/.

Anyway this does not require immediate attention and could be discussed later.

**Pull request merged**
